project3/coordinator/registrar.h

```diff
@@ -0,0 +1,52 @@
+/**
+ * @file Registrar to be utilized by the coordinator.
+ */
+#ifndef PROJECT3_REGISTRAR_H
+#define PROJECT3_REGISTRAR_H
+#include "connected_participants.h"
+#include <memory>
+
+namespace coordinator::registrar {
+using namespace connected_participants;
```

Comment:
Don't use `using namespace` declarations. Import each definition that you need individually, e.g. `using connected_participants::MyType`.